core/src/allocation.jl

```diff
@@ -818,7 +899,12 @@ function allocate!(p::Parameters, allocation_model::AllocationModel, t::Float64)
         # or set edge capacities if priority_idx = 1
         adjust_edge_capacities!(allocation_model, priority_idx)
 
-        set_demands_priority!(allocation_model, user, priority_idx, t)
+        # Set the objective depending on the demands
+        # A new objective function is set instead of modifying the coefficients
+        # of an existing objective function because this is not supported for
+        # quadratic terms:
+        # https://jump.dev/JuMP.jl/stable/manual/objective/#Modify-an-objective-coefficient
```

Comment:
```suggestion
        # https://jump.dev/JuMP.jl/v1.16/manual/objective/#Modify-an-objective-coefficient
```

Link is less likely to die this way.